cmd/dockerd/daemon.go

```diff
@@ -248,6 +248,12 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 		return fmt.Errorf("Error starting daemon: %v", err)
 	}
 
+	if d.HasExperimental() && cli.Config.MetricsAddress != "" {
```

Comment:
let's split it a little and return an error if `MetricsAddress` is not empty on non-experimental.